### PR TITLE
Add a day-of-week effect to simulated dataset

### LINKED ISSUE #90: Simulated day-of-week effect
Add a simulated day of week effect to the simulated data timeseries.

I'm pushing handling a full data simulator to #42.

### PR DESCRIPTION
**Pull request by zsusswein**
The current timeseries doesn't have a day of week effect, which makes it hard to test how the day of week effect implementation is performing.
This modifies the simulated timeseries to add a large-ish day of week reporting effect.

In implementing this change, I converted the timeseries over to using a `data-raw` script for reproducibility.
I moved the original Gostic "Practical considerations" dataset into an RDS object in `data-raw`.
This change makes the dataset tracked and available for re-generation of the package dataset, but doesn't expose it to or ship it with the package.
It's not 100% reproducible because I don't implement the git clone of the "Practical considerations" repo, but I think it gets us a fair but further than before.

I also clean up the package-provided `stochastic_sir_rt` datset.
I rename the `obs_incidence` col to `obs_cases`, add a `obs_cases_no_dow` for testing, and rename `date` to `reference_date` in anticipation of a future `report_date`.

Note that this change causes the non-day-of-week model performance to decrease in the vignette.
I'm expecting this change will be temporary until we can implement #89, which this PR unblocks.

Closes #90

### CODE REVIEW

File: R/data.R [hunk 1]
```diff
@@ -25,33 +28,28 @@
...
-#'     `incidence` column with added negative-binomial observation noise.
-#'     Created with `set.seed(123456)` and the call
-#'      `rnbinom(299, mu = [["incidence"]], size = 10)` Useful for
-#'       testing.}
-#'    \item{true_r0}{The initial R0 of the system (i.e., 2)}
-#'    \item{true_rt}{The known, true Rt of the epidemic system}
+#'    \item{reference_date}{The date the cases were observed.}
+#'    \item{true_rt}{The known, true Rt of the epidemic system.}
+#'    \item{dow}{The magnitude of the day-of-week effect added to the log of `incidence`.}
+#'    \item{incidence}{The true number of cases occurring on the `date` in the
```
Comment (kgostic) [inline R/data.R:39]:
```suggestion
#'    \item{incidence}{The true number of infections occurring on the `date` in the
```

Is this correct?

Comment (kgostic) [reply R/data.R:39]: It looks like you dropped some columns here. Can you go back and make sure the documentation is clear about infections vs. cases (defined as infections convolved with a delay). I'd also consider whether you want DOW effects to modify infections or cases.

**Review by kgostic — APPROVED** [review]
I think this looks reasonable but please double check the infections vs. cases language and conceptual choices!

**Comment by zsusswein** [comment]
Hm yeah I'll take another pass at the docs.

For the simulated data, we've been entirely ignoring infection to case observation delays for now. See, for example, the predict method in the vignette: https://cdcgov.github.io/cfa-gam-rt/articles/RtGam.html

Although not explored vs simulated data, we currently implement just a mean-shift as a correction. I have it on the roadmap to explore that more vs simulated data in #42 and we're discussing potential different approaches for the model in #88.

**Comment by kgostic** [comment]
[like]  Gostic, Katelyn (CDC/OD/ORR/CFA) reacted to your message:
________________________________

**Review by seabbs — APPROVED** [review]
LGTM once the suggestions from @kgostic are actioned.

File: R/data.R (hunk 1 above)
Comment (zsusswein) [reply R/data.R]: https://github.com/CDCgov/cfa-gam-rt/pull/91/commits/90f5f700e686fcd473deca09ab3db43fa7c0b7fe

**Pull request merged**